### ISSUE TITLE
Deposit form:  "major" accessibility issues that may not need design

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/happy-heron#3170]
From https://docs.google.com/spreadsheets/d/1Fs5sY1zgpovWVdHMqpJWRu2Zvyr53jw7, here are "major" accessibility issues for the **Deposit Form** that we can fix without design changes.  Note that I addressed accessibility level A and AA problems noted by the SiteImprove, WAVE and Axe browser plugins in PR #3169.

Note that solutions may be suggested in https://docs.google.com/spreadsheets/d/1Fs5sY1zgpovWVdHMqpJWRu2Zvyr53jw7.

These can be separate PRs, and/or split into separate tickets.

- [ ] ~~There are multiple "Remove" buttons that perform different actions.  Form field labels should be unique.~~ See #3181
- [x] When the "Upload one or more files (more than 10GB) which will be displayed as uploaded, including any file hierarchy" option  is selected, the six steps listed are not structured as a list. See #3333
- [ ] When the "See more options" control is activated, focus moves past the new content to the "Continue" button (in both the modal dialog and in the main page).
- [x] When the "See more options"/"See fewer options" disclosure widget is used it is missing an aria-expanded (in the modal dialog and on the main page).
- [x] There are headings that are only styled to look like headings but are not marked as such in HTML.  Each section has a visible heading, but they are not marked up as headings.  Examples include:
    - Add your files*
    - Title of deposit and contact information*
    - List authors and contributors *
and many more.
- [x] There are groups of form controls that do not provide <fieldset> and <legend> elements.  The radio buttons in "Add your files*" and the date fields in "Enter dates related to your deposit" are not grouped appropriately. 
- [ ] Form field constraints and/or errors are either not clearly displayed or are not programmatically associated with the field they apply to.  Errors are not explicitly associated with form fields, which means that the error text is not conveyed when the fields have focus and are in error.   
        There are `<select>` elements with asterisk that are not programmatically conveyed as required: 
          - In the List authors and contributors section, the Role term `<select>`  
          - In the Select a license section, the License `<select>`
- [x] The toggle for "Use auto-generated citation" does not receive keyboard focus: PR #3335
- [x] The "Remove" buttons (trash can images) receive keyboard focus with no visual indication when using TAB or SHIFT+TAB to navigate through them. PR #3335
- [x] There are numerous instances of placeholder links, <a> elements without an href.  However, they are receiving focus because tabindex="0" has been added which means they are not acting as placeholder links.  Each of these lacks a role.  Custom controls should provide textual name (link or button label, form label text), role (i.e. identify as button, link, text field, etc.), and/or state information (i.e. "expanded", "collapsed", selected, current).  Possible fix shown in PR #3161;  Same problem listed in #3176
- [ ] the 4 errors noted by WAVE (I could not find the elements to fix)
    ![image](https://github.com/sul-dlss/happy-heron/assets/96775/d4210248-feeb-4c43-8659-a03c2db7f914)
- [ ] (Minor level) The aria-expanded attribute is not allowed on a `<div>` element.

**Comment by ndushay** [comment]
![image.png](https://images.zenhubusercontent.com/5d89272aae87b600015057e7/0653ced3-5171-4c3e-a06c-8e3f6df7af2c)
